cql3/statements/schema_altering_statement.hh

```diff
@@ -62,7 +62,9 @@ protected:
     execute(query_processor& qp, service::query_state& state, const query_options& options, std::optional<service::group0_guard> guard) const override;
 
 public:
-    virtual future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>> prepare_schema_mutations(query_processor& qp, const query_options& options, api::timestamp_type) const = 0;
+    using event_t = cql_transport::event::schema_change;
+    virtual future<std::tuple<::shared_ptr<event_t>, std::vector<mutation>, cql3::cql_warnings_vec>> prepare_schema_mutations(query_processor& qp, const query_options& options, api::timestamp_type) const = 0;
+    virtual ::shared_ptr<event_t> created_event() const { return nullptr; }
```

Comment:
Why does this need to be virtual? IIUC all calls access the implementation in the same class.

Reply:
some use derived implementation (namely ks/table/type/view statements)

Reply:
virtual call in `cql3: always return created event in create ks/table/type/view statement`  commit